packages/runtime/src/components/hooks/useRouterLocaleSync.ts

```diff
@@ -1,16 +1,27 @@
-import { useRouter } from 'next/router'
-
+import { useRouter as usePagesRouter } from 'next/router'
+import { P, match } from 'ts-pattern'
 import { useEffect } from 'react'
+
 import { setLocale } from '../../state/actions'
 import { useDispatch } from '../../runtimes/react/hooks/use-dispatch'
 
+function useRouter() {
+  try {
+    const router = usePagesRouter()
+
+    return router
+  } catch (e) {
+    return;
```

Comment:
And of course, this isn't explicit, we're implicitly returning `undefined`, but whatever calls this might have to handle `undefined`. Again, this is a major nit, so feel free to ignore.

Reply:
We could easily go into a debate here about this 😅 I'm going to leave it for now and if it needs to change, we can update it in another PR.